pallets/connectors/src/message.rs

```diff
@@ -201,6 +201,90 @@ where
 		tranche_id: TrancheId,
 		investor: Address,
 	},
+	/// The message sent back to the domain from which a `DecreaseInvestOrder`
+	/// message was received, ensuring the correct state update on said domain
+	/// and that the `investor`'s wallet is updated accordingly.
+	///
+	/// Directionality: Centrifuge -> EVM Domain.
+	ExecutedDecreaseInvestOrder {
+		/// The pool id
+		pool_id: PoolId,
+		///The tranche id
+		tranche_id: TrancheId,
+		/// The investor's address
+		investor: Address,
+		/// The currency in which `DecreaseInvestOrder` was realised
+		currency: u128,
+		/// The amount of `currency` that was actually executed in the original
+		/// `DecreaseInvestOrder` message, i.e., the amount by which the
+		/// investment order was actually decreased by.
+		decreased_by: Balance,
+		/// The outstanding order, in `currency` units
+		outstanding_order: Balance,
+	},
+	/// The message sent back to the domain from which a `DecreaseRedeemOrder`
+	/// message was received, ensuring the correct state update on said domain
+	/// and that the `investor`'s wallet is updated accordingly.
+	///
+	/// Directionality: Centrifuge -> EVM Domain.
+	ExecutedDecreaseRedeemOrder {
+		/// The pool id
+		pool_id: PoolId,
+		///The tranche id
+		tranche_id: TrancheId,
+		/// The investor's address
+		investor: Address,
+		/// The currency in which `DecreaseRedeemOrder` was realised
+		currency: u128,
+		/// The amount of `currency` that was actually executed in the original
+		/// `DecreaseRedeemOrder` message, i.e., the amount by which the order
+		/// was actually decreased by.
+		decreased_by: Balance,
+		/// The outstanding order, in `currency` units
+		outstanding_order: Balance,
+	},
+	/// The message sent back to the domain from which a `CollectInvest` message
+	/// has been received, which will ensure the `investor` gets the payout
+	/// respective to their investment.
+	///
+	/// Directionality: Centrifuge -> EVM Domain.
+	ExecutedCollectInvest {
+		/// The pool id
+		pool_id: PoolId,
+		/// The tranche
+		tranche_id: TrancheId,
+		/// The investor's address
+		investor: Address,
+		/// The currency in which the investment was realised
+		currency: u128,
+		/// The amount that was actually invested
+		amount_invested: Balance,
+		/// The amount of tranche tokens received for the investment made
+		tranches_received: Balance,
+		/// The remaining amount of `currency` the investor still has locked to
+		/// invest at a later epoch execution
+		remaining_locked_investment: Balance,
+	},
+	/// The message sent back to the domain from which a `CollectRedeem` message
+	/// has been received, which will ensure the `investor` gets the payout
+	/// respective to their redemption.
+	///
+	/// Directionality: Centrifuge -> EVM Domain.
+	ExecutedCollectRedeem {
```

Comment:
Based on

```
ExecutedCollectRedeem
 -  How many shares (TIN/DROP) were actually redeemed
 -  How much currency (stable coins) was received for the redeemed shares
 -  optional: currencyID
 -  How many shares left to be redeemed (could not be included in the epoch execution)
```

Reply:
I think 
 -  How many shares (TIN/DROP) were actually redeemed
 
is missing in this message.

Reply:
I sort of remember @ilinzweilin saying that that's something we could calculate on the EVM side by having the `payout` amount and the price of the tranche token. But I guess that just having that info in the message saves computation, fees, and margin of error on the EVM side.